src/Repository/ScheduledContentRepositoryTrait.php

```diff
@@ -7,37 +7,40 @@
 trait ScheduledContentRepositoryTrait
 {
     /**
-     * Usage: $queryBuilder->addCriteria()
+     * Usage: $queryBuilder->addCriteria(self::getPublishedCriteria())
      */
-    private function generatePublisedWhereClausesCriteria(string $alias = 'p', \DateTimeInterface $refDate = null): Criteria
+    protected static function getPublishedCriteria(?string $alias = 'p', \DateTimeInterface $refDate = null): Criteria
     {
         if (null === $refDate) {
             $refDate = new \DateTimeImmutable('now');
         }
+
+        $prefix = $alias !== null && $alias !== '' ? $alias . '.' : '';
+
         $expr = Criteria::expr();
 
         return Criteria::create()
             ->andWhere(
-                $expr->eq($alias . '.isPublic', true)
+                $expr->eq($prefix . 'isPublic', true)
             )->andWhere(
-                $expr->orX(
-                    $expr->andX(
-                        $expr->isNull($alias . '.dateScheduledFrom'),
-                        $expr->isNull($alias . '.dateScheduledTill'),
+                $expr->andX(
+                    $expr->orX(
+                        $expr->isNull($prefix . 'dateScheduledFrom'),
+                        $expr->lte($prefix . 'dateScheduledFrom', $refDate)
                     ),
-                    $expr->andX(
-                        $expr->lte($alias . '.dateScheduledFrom', $refDate),
-                        $expr->isNull($alias . '.dateScheduledTill')
-                    ),
-                    $expr->andX(
-                        $expr->isNull($alias . '.dateScheduledFrom'),
-                        $expr->gte($alias . '.dateScheduledTill', $refDate)
-                    ),
-                    $expr->andX(
-                        $expr->lte($alias . '.dateScheduledFrom', $refDate),
-                        $expr->gte($alias . '.dateScheduledTill', $refDate)
+                    $expr->orX(
+                        $expr->isNull($prefix . 'dateScheduledTill'),
+                        $expr->gte($prefix . 'dateScheduledTill', $refDate)
                     )
                 )
             );
     }
+
+    /**
+     * @deprecated Use static call instead: self::getPublishedCriteria()
+     */
+    private function generatePublisedWhereClausesCriteria(string $alias = 'p', \DateTimeInterface $refDate = null): Criteria
+    {
+        return self::getPublishedCriteria($alias, $refDate);
```

Comment:
Niet relevant, maar hier zou `static::` volgens mij beter passen in geval iemand deze klasse extend, de methode `getPublishedCriteria` overschrijft en aanpast maar vervolgens toch `generatePublisedWhereClausesCriteria` aanroept :upside_down_face: